### PR TITLE
update per new setting java.jdt.ls.java.home

### PR DESCRIPTION
**Pull request by Eskibear**
per upstream change https://github.com/redhat-developer/vscode-java/pull/2284

Signed-off-by: Yan Zhang <yanzh@microsoft.com>

### CODE REVIEW

**Review by jdneo — APPROVED** [review]

File: src/java-runtime/utils/upstreamApi.ts [hunk 1]
```diff
@@ -42,51 +42,52 @@ export async function resolveRequirements(): Promise<{
             javaVersion = await getMajorVersion(javaHome);
             toolingJre = javaHome;
```
Comment (testforstephen) [inline src/java-runtime/utils/upstreamApi.ts:43]: On vscode-java, it's a different about picking java.home as tooling jre.

image
See [source code on vscode-java](https://github.com/redhat-developer/vscode-java/blob/de65a586d36a79034a533d564f470b3a2754c1fe/src/requirements.ts#L56-L59)

Comment (Eskibear) [reply src/java-runtime/utils/upstreamApi.ts:43]: so in vscode-java, `java.home` will not be picked up as jdtls runtime, right?

Comment (Eskibear) [reply src/java-runtime/utils/upstreamApi.ts:43]: BTW, here the logic is to determine whether to pop up "configure Java runtime" for the tooling runtime. Different from upstream vscode-java, here in pack, we simply detect whether toolingJre is valid. Default project runtime `javaHome` is not important.

Comment (testforstephen) [reply src/java-runtime/utils/upstreamApi.ts:43]: > so in vscode-java, java.home will not be picked up as jdtls runtime, right?

Yes, here is the order to look for a tooling JRE:
"java.jdt.ls.java.home" > embedded jre > "java.home"

Comment (Eskibear) [reply src/java-runtime/utils/upstreamApi.ts]: Thanks, updated.

**Review by testforstephen — APPROVED** [review]

**Pull request merged**